stock_inventory_revaluation/models/stock_inventory_revaluation.py

```diff
@@ -0,0 +1,407 @@
+# -*- coding: utf-8 -*-
+# © 2015 Eficent Business and IT Consulting Services S.L.
+# - Jordi Ballester Alomar
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from openerp import api, fields, models, _
+import openerp.addons.decimal_precision as dp
+import time
+from openerp.exceptions import Warning as UserError
+
+_STATES = [
+    ('draft', 'Draft'),
+    ('posted', 'Posted'),
+    ('cancel', 'Cancelled')]
+
+
+class StockInventoryRevaluation(models.Model):
+
+    _name = 'stock.inventory.revaluation'
+    _description = 'Inventory revaluation'
+
+    @api.model
+    def _default_journal(self):
+        res = self.env['account.journal'].search([('type', '=', 'general')])
+        return res and res[0] or False
+
+    @api.one
+    def _get_product_template_qty(self):
+        self.qty_available = 0
+        for prod_variant in self.product_template_id.product_variant_ids:
+            self.qty_available += prod_variant.qty_available
+
+    @api.one
+    def _calc_product_template_value(self):
+        qty_available = 0
+        current_value = 0.0
+        quant_obj = self.env['stock.quant']
+        for prod_variant in self.product_template_id.product_variant_ids:
+            qty_available += prod_variant.qty_available
+            if self.product_template_id.cost_method == 'real':
+                quants = quant_obj.search([('product_id', '=',
+                                            prod_variant.id),
+                                           ('location_id.usage', '=',
+                                            'internal')])
+                for quant in quants:
+                    current_value += quant.cost
+            else:
+                current_value = \
+                    self.product_template_id.standard_price * qty_available
+        self.current_value = current_value
+
+    name = fields.Char('Reference',
+                       help="Reference for the journal entry",
+                       readonly=True,
+                       required=True,
+                       states={'draft': [('readonly', False)]},
+                       copy=False,
+                       default='/')
+
+    revaluation_type = fields.Selection(
+        [('price_change', 'Price Change'),
+         ('inventory_value', 'Inventory Debit/Credit')],
+        string="Revaluation Type",
+        readonly=True, required=True,
+        default='price_change',
+        help="'Price Change': You can re-valuate inventory values by Changing "
+             "the price for a specific product. The inventory price is "
+             "changed and inventory value is recalculated according to the "
+             "new price.\n "
+             "'Inventory Debit/Credit': Changing the value of the inventory. "
+             "The quantity of inventory remains unchanged, resulting in a "
+             "change in the price",
```

Comment:
Not sure this is clear enough. One method lets you enter the unit price and the other lets you enter the total value?